internal/leadership/elector.go

```diff
@@ -567,27 +568,38 @@ func attemptElectOrReelect(ctx context.Context, exec riverdriver.Executor, alrea
 	ctx, cancel := context.WithTimeout(ctx, deadlineTimeout)
 	defer cancel()
 
-	return dbutil.WithTxV(ctx, exec, func(ctx context.Context, exec riverdriver.ExecutorTx) (bool, error) {
-		if _, err := exec.LeaderDeleteExpired(ctx, &riverdriver.LeaderDeleteExpiredParams{
-			Now:    params.Now,
-			Schema: params.Schema,
-		}); err != nil {
-			return false, err
+	execTx, err := exec.Begin(ctx)
+	if err != nil {
+		var additionalDetail string
+		if errors.Is(err, context.DeadlineExceeded) {
+			additionalDetail = " (a common cause of this a database pool that's at its connection limit; you may need to increase maximum connections)"
```

Comment:
```suggestion
			additionalDetail = " (a common cause of this is a database pool that's at its connection limit; you may need to increase maximum connections)"
```

Also, this brings to mind that we really need a doc section on this because it'd be amazing to link to that from here. Could even adopt a shortened permalink redirect convention like `riverqueue.com/e/elector-pool-size` to keep working links even as we reorganize docs.

I can start on this page later today unless you want to take that too.